test-infra/prowjobs/cleanerupper/main.go

```diff
@@ -359,6 +362,48 @@ func cleanGuestPolicies(ctx context.Context, client *osconfig.Client, project st
 	wg.Wait()
 }
 
+func cleanOSPolicyAssignments(ctx context.Context, computeClient daisyCompute.Client, client *osconfigV1alpha.OsConfigZonalClient, project string) {
+	fmt.Println("Cleaning OSPolicyAssignments:")
+	var wg sync.WaitGroup
+	zones, err := computeClient.ListZones(project)
+	if err != nil {
+		fmt.Printf("Error calling ListZones in project %q: %v\n", project, err)
+		return
+	}
+	for _, zone := range zones {
+		now := time.Now()
+		itr := client.ListOSPolicyAssignments(ctx, &osconfigv1alphapb.ListOSPolicyAssignmentsRequest{Parent: fmt.Sprintf("projects/%s/locations/%s", project, zone.Name)})
+		for {
+			ospa, err := itr.Next()
+			if err != nil {
+				if err == iterator.Done {
+					fmt.Println(time.Since(now), zone.Name)
+					break
+				}
+				fmt.Printf("Error calling ListOSPolicyAssignments in project %q: %v\n", project, err)
+				return
+			}
+			if !shouldDelete(ospa.GetName(), nil, ospa.GetDescription(), ospa.GetRevisionCreateTime().GetSeconds()) {
+				continue
+			}
+			if *dryRun {
+				continue
+			}
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				op, err := client.DeleteOSPolicyAssignment(ctx, &osconfigv1alphapb.DeleteOSPolicyAssignmentRequest{Name: ospa.GetName()})
+				if err != nil {
+					fmt.Printf("Error deleting OSPolicyAssignment: %v\n", err)
+					return
+				}
+				op.Wait(ctx)
```

Comment:
Will this return when you get 404 not found which is the current behavior after the assignment is deleted

Reply:
Yes, this will wait for operation to be Done, which will never happen as the operation will disappear, so Wait will return an error that we do not check for, so this works fine for this use case.